### PR TITLE
Implementation of sync service

### PR DESCRIPTION
**Pull request by sitaram-kalluri**
**- What I did** Initial implementation of sync service.

TODO's : 1. Implement getLatestServerCommitId via stats notification service. 
                2. lastSyncedOn in SyncResult. Had an approach. Need a discussion.

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: at_client/lib/src/service/sync_service.dart [hunk 1]
```diff
@@ -1,376 +1,161 @@
...
-      var syncObject = await _getSyncObject(regex: regex);
-      var lastSyncedCommitId = syncObject.lastSyncedCommitId;
-      var serverCommitId = syncObject.serverCommitId;
-      var isInSync = SyncUtil.isInSync(syncObject.uncommittedEntries,
-          syncObject.serverCommitId, syncObject.lastSyncedCommitId);
-      if (isInSync) {
-        _logger.finer('Server and local secondary are in sync');
-        _syncInProgress = false;
+      // Check if local and cloud secondary are in sync. If true, return.
+      if (await _isInSync()) {
```
Comment (VJag) [inline at_client/lib/src/service/sync_service.dart]: Do we need this in a try catch?

File: at_client/lib/src/service/sync_service.dart [hunk 2]
```diff
@@ -1,376 +1,161 @@
...
+        syncResult = SyncResult();
```
Comment (VJag) [inline at_client/lib/src/service/sync_service.dart]: Should we not set attributes on syncResult bean?

Comment (VJag) [reply at_client/lib/src/service/sync_service.dart]: Do we need a lastSynedOn?

File: at_client/lib/src/service/sync_service.dart [hunk 3]
```diff
@@ -1,376 +1,161 @@
...
-    var commitEntry = await SyncUtil.getCommitEntry(sequenceNumber, _atSign);
-    commitEntry!.operation = operation;
-    await SyncUtil.updateCommitEntry(
-        commitEntry, serverCommitEntry['commitId'], _atSign);
-    _lastSyncedTime = DateTime.now();
-    _lastSyncedCommitId = serverCommitEntry['commitId'];
-    if (serverCommitEntry['commitId'] == _serverCommitId) {
-      _change--;
+      // Get latest server commit id.
+      serverCommitId = await _getServerCommitId();
```
Comment (VJag) [inline at_client/lib/src/service/sync_service.dart]: This is happening in isInSync too? Will improve _getServerCommitId later.

File: at_client/lib/src/service/sync_service.dart (hunk 1 above)
Comment (sitaram-kalluri) [reply at_client/lib/src/service/sync_service.dart]: This is in try catch block.

**Review by murali-shris — APPROVED** [review]

**Pull request merged**